src/box_utils.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2021, SmileRobotics, Japan
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   1. Redistributions of source code must retain the above
+ *      copyright notice, this list of conditions and the following
+ *      disclaimer.
+ *
+ *   2. Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following
+ *      disclaimer in the documentation and/or other materials provided
+ *      with the distribution.
+ *
+ *   3. Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived
+ *      from this software without specific prior written permission.
+ *
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ *  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ *  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ *  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include <geometry_msgs/Point32.h>
+#include <ros/ros.h>
+
+#include "box_utils.h"
+
+std::string boxToString(const Box& box)
+{
+  std::string box_string = "[";
+
+  box_string +=
+      "[" + std::to_string(box.min.x) + ", " + std::to_string(box.min.y) + ", " + std::to_string(box.min.z) + "]";
+
+  box_string += ", ";
+
+  box_string +=
+      "[" + std::to_string(box.max.x) + ", " + std::to_string(box.max.y) + ", " + std::to_string(box.max.z) + "]";
+
+  box_string += "]";
+  return box_string;
+}
+
+Box makeBoxFromTwoPoints(const std::vector<geometry_msgs::Point32>& points)
+{
+  Box box;
+  if (points.size() == 2)
+  {
+    box.min.x = std::min(points[0].x, points[1].x);
+    box.min.y = std::min(points[0].y, points[1].y);
+    box.min.z = std::min(points[0].z, points[1].z);
+
+    box.max.x = std::max(points[0].x, points[1].x);
+    box.max.y = std::max(points[0].y, points[1].y);
+    box.max.z = std::max(points[0].z, points[1].z);
+  }
```

Comment:
I think it seems better to return some error information if the input vector size is not two.
How about adding ROS_ERROR or ROS_WARN?

Reply:
You are right.
I have re-designed the function in [here](https://github.com/smilerobotics/laser_filters/pull/5/commits/d58c6c97442f0a2c48e115fe6ac25c6ab4726fa5).